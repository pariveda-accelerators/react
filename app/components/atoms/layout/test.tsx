import { render } from 'react-testing-library';
import { InjectElement, Spacer, Column, Row, JustifyContent, BgColor } from './';
import React from 'react';

const getChildren = () => (
  <>
    <div key={1}>One</div>
    <div key={2}>Two</div>
    <div key={3}>Three</div>
  </>
);

const getColumn = (
  justification: JustifyContent = null,
  childSpacing: number = 0,
  bgColor: BgColor = null,
) => (
  <Column
    justifyContent={justification}
    childSpacing={childSpacing}
    bgColor={bgColor}
  >
    {getChildren()}
  </Column>
);

const getRow = (
  justification: JustifyContent = null,
  childSpacing: number = 0,
  bgColor: BgColor = null,
) => (
  <Row justifyContent={justification} childSpacing={childSpacing} bgColor={bgColor}>
    {getChildren()}
  </Row>
);

describe('Layout Atoms', () => {
  describe('Inject Element', () => {
    test('Injects Correct Number of Elements', () => {
      const { getAllByTestId, container } = render(
        <InjectElement element={<hr data-testid="hr" />}>
          <div key={1}>hello</div>
          <div key={2}>hello</div>
          <div key={3}>hello</div>
        </InjectElement>,
      );
      expect(getAllByTestId('hr').length).toEqual(2);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Spacer', () => {
    test('Renders Width', () => {
      const { container } = render(<Spacer width={16} />);
      expect(container).toMatchSnapshot();
    });
    test('Renders Height', () => {
      const { container } = render(<Spacer height={16} />);
      expect(container).toMatchSnapshot();
    });
    test('Renders Width + Height', () => {
      const { container } = render(<Spacer width={16} height={16} />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('Column', () => {
    test('Renders Column', () => {
      const { container } = render(getColumn());
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Child Spacing', () => {
      const { container } = render(getColumn(null, 16, null));
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Background Color - WHITE', () => {
      const { container } = render(getColumn(null, 16, BgColor.WHITE));
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Background Color - GREY', () => {
      const { container } = render(getColumn(null, 16, BgColor.GREY));
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Background Color - BLUE', () => {
      const { container } = render(getColumn(null, 16, BgColor.BLUE));
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Content Justification - START', () => {
      const { container } = render(getColumn(JustifyContent.START));
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Content Justification - CENTER', () => {
      const { container } = render(getColumn(JustifyContent.CENTER));
      expect(container).toMatchSnapshot();
    });
    test('Renders Column with Content Justification - END', () => {
      const { container } = render(getColumn(JustifyContent.END));
      expect(container).toMatchSnapshot();
    });
  });

  describe('Row', () => {
    test('Renders Row', () => {
      const { container } = render(getRow());
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Child Spacing', () => {
      const { container } = render(getRow(null, 16, null));
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Background Color - WHITE', () => {
      const { container } = render(getRow(null, 16, BgColor.WHITE));
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Background Color - GREY', () => {
      const { container } = render(getRow(null, 16, BgColor.GREY));
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Background Color - BLUE', () => {
      const { container } = render(getRow(null, 16, BgColor.BLUE));
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Content Justification - START', () => {
      const { container } = render(getRow(JustifyContent.START));
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Content Justification - CENTER', () => {
      const { container } = render(getRow(JustifyContent.CENTER));
      expect(container).toMatchSnapshot();
    });
    test('Renders Row with Content Justification - END', () => {
      const { container } = render(getRow(JustifyContent.END));
      expect(container).toMatchSnapshot();
    });
  });
});
