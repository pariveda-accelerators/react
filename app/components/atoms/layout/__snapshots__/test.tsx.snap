// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Atoms Column Renders Column 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Background Color - BLUE 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #3867d6;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Background Color - GREY 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #D3D9E1;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Background Color - WHITE 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #f5f6fa;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Child Spacing 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Content Justification - CENTER 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Content Justification - END 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Column Renders Column with Content Justification - START 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Inject Element Injects Correct Number of Elements 1`] = `
<div>
  <div>
    hello
  </div>
  <hr
    data-testid="hr"
  />
  <div>
    hello
  </div>
  <hr
    data-testid="hr"
  />
  <div>
    hello
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Background Color - BLUE 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #3867d6;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Background Color - GREY 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #D3D9E1;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Background Color - WHITE 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #f5f6fa;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Child Spacing 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Content Justification - CENTER 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Content Justification - END 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Row Renders Row with Content Justification - START 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: transparent;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      One
    </div>
    <div>
      Two
    </div>
    <div>
      Three
    </div>
  </div>
</div>
`;

exports[`Layout Atoms Spacer Renders Height 1`] = `
.c0 {
  height: 16px;
}

<div>
  <div
    class="c0"
    height="16"
  />
</div>
`;

exports[`Layout Atoms Spacer Renders Width + Height 1`] = `
.c0 {
  width: 16px;
  height: 16px;
}

<div>
  <div
    class="c0"
    height="16"
    width="16"
  />
</div>
`;

exports[`Layout Atoms Spacer Renders Width 1`] = `
.c0 {
  width: 16px;
}

<div>
  <div
    class="c0"
    width="16"
  />
</div>
`;
